### ISSUE TITLE
Possible data inconsistency in quest joining

### ISSUE THREAD

**Issue by mbanys** [HabitRPG/habitica#6621]
I (user ID 5a74ce44-d856-4323-b522-db126b85bba6) started a boss quest for my party (Group ID: 7a85c791-cc11-4d8a-8092-c0e60ccf6ce4) on tuesday, Jan 26. I invited all 3 of my party members.  Shortly thereafter, two of us got the Quest Started email, while the third person (goddess1903 in my party -- sorry, I don't have her User ID) did not.  However, she DID join the quest from the invite email she received, and the quest started itself, which tells me that all party members accepted the invitation, thus auto-starting the quest).  

In the iOS app, in the Quest section of the Party screen, it says there are 3 Participants in the quest, and when I click on that, all three of us are shown as having Accepted the quest.  However, only the two of us that received Quest Started emails are dealing damage.  In the Desktop version of Habitica, only the two of us are shown as Participants in the quest.

Therefore, there seems to be a data inconsistency between the two platforms.  I know this third member performed the actions necessary to join the quest (she's done it correctly before, the quest auto-started within a few hours of my invitation, and iOS shows 3 Participants).  However, Desktop seems to reflect that only two of us accepted the quest, which is consistent with how Habitica is dealing damage to the boss only by us two.

It's possible it could be a sync error; however, it's worth posting details here to help reduce this issue going forward.

Thanks,
Mike

PS the desktop version I'm seeing this on is latest chrome on both windows 7 and windows 8.  iOS version is the latest as well, on both an iPad and an iPhone.

**Comment by Alys** [comment]
Can you please tell us goddess1903's User ID? You can find it by clicking on her profile and it will be on the left had side not far from the top, or you can ask her what it is - she can find it from Settings -> API.

**Comment by mbanys** [comment]
Sure, here it is:  6dad373e-8211-4da2-9872-85bdb94694ed.

**Comment by Alys** [comment]
Based on what I seen now in goddess1903's account in the database and in the party's data, goddess1903 rejected the quest invitation rather than accepted it. 

> two of us got the Quest Started email, while the third person (goddess1903 in my party -- sorry, I don't have her User ID) did not.

That's another sign that she had rejected. Those emails are sent to only players who accepted the quest.

> However, she DID join the quest from the invite email she received, and the quest started itself, which tells me that all party members accepted the invitation, thus auto-starting the quest).

A quest can also auto-start if some players reject it. It starts as soon as every player has either accepted or rejected.

> In the iOS app, in the Quest section of the Party screen, it says there are 3 Participants in the quest, and when I click on that, all three of us are shown as having Accepted the quest.

That sounds like a bug in the iOS app. Could you please take a screenshot and add it to this issue? You'll need to visit GitHub (https://github.com/HabitRPG/habitrpg/issues/6621 ) and drag the image into a new comment. Attaching it to an email won't work. I'll then open a matching issue in the iOS repository.

> However, only the two of us that received Quest Started emails are dealing damage. In the Desktop version of Habitica, only the two of us are shown as Participants in the quest.

Both of those are the correct behaviour for this case.

> I know this third member performed the actions necessary to join the quest (she's done it correctly before

This time she might have accidentally clicked on the wrong button. It's also possible that there's a bug which sometimes records an acceptance as a rejection, but I don't think we've seen other reports of that (which is not to say it doesn't happen; it would just be rare).

Basically, what you're seeing on the website and in the quest damage messages is correct: goddess1903 is not a quest participant. If you wish to, you could abort the quest. The scroll would go back into your inventory for reuse.

**Comment by mbanys** [comment]
Thanks Alys for the fast feedback!  It is possible she clicked the wrong button--that's what I would have figured had the iOS app not reported conflicting information.  I've attached two screenshots from the Party window as requested.
![2016-01-30 12 14 12](https://cloud.githubusercontent.com/assets/16975843/12697504/5f1f150e-c74b-11e5-872b-cae482eb69ba.jpg)
![2016-01-30 12 14 05](https://cloud.githubusercontent.com/assets/16975843/12697505/6308f400-c74b-11e5-9732-595a7640c50e.jpg)

**Comment by Alys** [comment]
Thanks for that! I've logged this as an iOS issue at https://github.com/HabitRPG/habitrpg-ios/issues/329 and linked back to this one. I'm closing this one now since there's nothing we can do about it here. Note though that you weren't wrong to report it here - it wasn't clear immediately that it was an iOS bug!
